core/src/main/resources/error/error-classes.json

```diff
@@ -118,14 +150,32 @@
     "message" : [ "PARTITION clause cannot contain a non-partition column name: %s" ],
     "sqlState" : "42000"
   },
+  "NOT_RESOLVED" : {
```

Comment:
In retrospect, this is specific to custom collection class resolution. Can we make this class name more specific? Eg. `CUSTOM_COLLECTION_CLASS_NOT_RESOLVED`